tests/testthat/test-as_tibble.R

```diff
@@ -735,3 +735,14 @@ test_that("output test", {
     as_tibble_row(list(a = 1:3, b = 1:3))
   })
 })
+
+# utilise as.data.frame for extended data.frames
+test_that("as_tibble.data.frame coerces extended data.frames first", {
+  x <- structure(mtcars, extra = "extra", class = c("ext_df_", "data.frame"))
+  y <- as_tibble(head(mtcars))
+  with_mocked_bindings(
+    as.data.frame = function(x, row.names = NULL, optional = FALSE, ...) y,
+    code = expect_identical(as_tibble(x), y),
```

Comment:
Just from looking at the code here it feels it should fail because `y` is only a head of `mtcars` while `x` is complete `mtcars`.

Reply:
Why should it fail? Are you expecting `as_tibble(x)` to be identical to `mtcars` converted to a tibble?

Would the test would be clearer if we used two completely unrelated data frames for `x` and `y`?

Reply:
because `y` is filtered out with `head(mtcars)` 3 lines above while `x` contains complete `mtcars`

Reply:
But the tests succeed, and the example should also work when running it line by line.